src/transformers/models/timm_wrapper/configuration_timm_wrapper.py

```diff
@@ -60,10 +64,30 @@ def __init__(self, initializer_range: float = 0.02, do_pooling: bool = True, **k
 
     @classmethod
     def from_dict(cls, config_dict: Dict[str, Any], **kwargs):
+        label_names = config_dict.get("label_names", None)
+
+        # if no labels added to config, use imagenet labeller in timm
+        if label_names is None:
+            imagenet_subset = infer_imagenet_subset(config_dict)
+            if imagenet_subset:
+                dataset_info = ImageNetInfo(imagenet_subset)
+                synsets = dataset_info.label_names()
+                label_descriptions = dataset_info.label_descriptions(as_dict=True)
+                label_names = [label_descriptions[synset] for synset in synsets]
+
+        is_custom_model = "num_labels" in kwargs or "id2label" in kwargs
+        if label_names is not None and not is_custom_model:
```

Comment:
If is_custom_model == True, shouldn't the imagenet labeller block above be skipped?

Reply:
Thanks, added